# Demonstrando a Importância da Vectorização (Vectorization)

## Comparando duas formas de vectorização (uma sem utilizar numpy e outra utilizando numpy)

In [2]:
a = np.array([1,2,3,4])

In [3]:
print(a)

[1 2 3 4]


In [4]:
import time
a = np.random.rand(1000000)

In [18]:
#Gerando número aleatórios e vectorização com numpy
b = np.random.rand(1000000)
tic  = time.time()
c = np.dot(a, b)
toc = time.time()

print(c)
print("Vectorized version = " + str(1000*(toc-tic)) + " ms")

250262.47152400124
Vectorized version = 1.508951187133789 ms


In [19]:
#Gerando sem numpy
c = 0
tic = time.time()
for i in range(1000000):
    c += a[i]*b[i]
toc = time.time()

print(c)
print("For loop version: " + str(1000*(toc-tic)) + "ms")

250262.47152399845
For loop version: 713.1078243255615ms


Como podemos observar o tempo que levou para computar com vectorization (numpy) foi de 1.508951187133789 ms e sem vectorization (sem numpy) foi de 713.1078243255615ms